.github/workflows/lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Lint
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "**.py"
+  pull_request:
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+          cache: "pip"
+      - name: Install dependencies
+        run: pip install -rrequirements{,-dev}.txt
+      - name: Run mypy
+        run: mypy .
```

Comment:
Also enabled "require status checks to pass before merging" and "require up to date branches before merging" in github 🎉